### PR TITLE
[Enhancement] Support Iceberg ScanMetrics for hive/glue catalog

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Planner profile looks like this:
截屏2023-03-30 19 51 28
and the scanMetricsReport looks like: it's defined by iceberg.
`- tt / No_1: ScanMetricsResult{totalPlanningDuration=TimerResult{timeUnit=NANOSECONDS, totalDuration=PT0.66060346S, count=1}, resultDataFiles=CounterResult{unit=COUNT, value=154}, resultDeleteFiles=CounterResult{unit=COUNT, value=0}, totalDataManifests=CounterResult{unit=COUNT, value=55}, totalDeleteManifests=CounterResult{unit=COUNT, value=0}, scannedDataManifests=CounterResult{unit=COUNT, value=55}, skippedDataManifests=CounterResult{unit=COUNT, value=0}, totalFileSizeInBytes=CounterResult{unit=BYTES, value=3524280}, totalDeleteFileSizeInBytes=CounterResult{unit=BYTES, value=0}, skippedDataFiles=CounterResult{unit=COUNT, value=0}, skippedDeleteFiles=CounterResult{unit=COUNT, value=0}, scannedDeleteManifests=CounterResult{unit=COUNT, value=0}, skippedDeleteManifests=CounterResult{unit=COUNT, value=0}, indexedDeleteFiles=CounterResult{unit=COUNT, value=0}, equalityDeleteFiles=CounterResult{unit=COUNT, value=0}, positionalDeleteFiles=CounterResult{unit=COUNT, value=0}}`

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java [hunk 1]
```diff
@@ -226,6 +241,7 @@ public static class Builder {
         private List<Column> fullSchema;
         private Map<String, String> icebergProperties;
         private org.apache.iceberg.Table nativeTable;
+        private IcebergMetricsReporter metricsReporter;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:244]: nit: Optional<IcebergMetricsReporter>

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -104,11 +110,20 @@ public Table getTable(String dbName, String tblName) {
         }
 
         try {
-            org.apache.iceberg.Table icebergTable = icebergCatalog.loadTable(identifier);
-            Table table = IcebergApiConverter.toIcebergTable(
-                    icebergTable, catalogName, dbName, tblName, icebergCatalog.getIcebergCatalogType().name());
-            tables.put(identifier, table);
-            return table;
+            if (icebergCatalog instanceof IcebergHiveCatalog || icebergCatalog instanceof IcebergGlueCatalog) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:113]: icebergCatalog.getIcebergCatalogType() == HIVE_CATALOG ...

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java [hunk 3]
```diff
@@ -44,6 +45,12 @@ public interface IcebergCatalog {
      */
     Table loadTable(TableIdentifier tableIdentifier) throws StarRocksConnectorException;
 
+    /**
+     * Loads a native Iceberg table based on ther information in 'feTable',
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java:49]: ther?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 4]
```diff
@@ -104,11 +110,20 @@ public Table getTable(String dbName, String tblName) {
...
+                IcebergMetricsReporter metricsReporter = new IcebergMetricsReporter();
+                org.apache.iceberg.Table icebergTable = icebergCatalog.loadTable(identifier, metricsReporter);
+                Table table = IcebergApiConverter.toIcebergTable(icebergTable, catalogName, dbName, tblName,
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:116]: Do we only support scanMetrics in the iceberg catalog ?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 5]
```diff
@@ -104,11 +110,20 @@ public Table getTable(String dbName, String tblName) {
...
+                        icebergCatalog.getIcebergCatalogType().name(), metricsReporter);
+                tables.put(identifier, table);
+                return table;
+            } else {
+                org.apache.iceberg.Table icebergTable = icebergCatalog.loadTable(identifier);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:121]: it's better to reduce some duplicate codes

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 6]
```diff
@@ -104,11 +110,20 @@ public Table getTable(String dbName, String tblName) {
...
+                Table table = IcebergApiConverter.toIcebergTable(
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:122]:
```suggestion
                IcebergCatalogType nativeType = icebergCatalog.getIcebergCatalogType();
            IcebergMetricsReporter metricsReporter = nativeType == HIVE_CATALOG || nativeType == GLUE_CATALOG ?
                    new IcebergMetricsReporter() : null;
            org.apache.iceberg.Table icebergTable = icebergCatalog.loadTable(identifier, metricsReporter);
            Table table = IcebergApiConverter.toIcebergTable(icebergTable, catalogName, dbName, tblName,
                    icebergCatalog.getIcebergCatalogType().name(), metricsReporter);
            tables.put(identifier, table);
            return table;
```

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 7]
```diff
@@ -175,11 +190,22 @@ private List<RemoteFileInfo> getRemoteFileInfos(IcebergTable table, long snapsho
...
                     builder.add(fileScanTask);
                 }
             }
+            try {
+                combinedScanTasks.close();
+            } catch (IOException e) {
+                // Ignored
+            }
+            table.reportScanMetrics().ifPresent(scanReportWithCounter -> {
+                PlannerProfile.addCustomProperties("Iceberg.Metadata.ScanMetrics." +
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:205]: Please provide a demo profile about iceberg in the problem summary.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 8]
```diff
@@ -175,11 +190,22 @@ private List<RemoteFileInfo> getRemoteFileInfos(IcebergTable table, long snapsho
...
+                                scanReportWithCounter.getScanReport().tableName() + " / No_" + scanReportWithCounter.getCount(),
+                        scanReportWithCounter.getScanReport().scanMetrics().toString());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:207]: My main concern was that there would be a huge number of metrics displayed

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 4 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:116]: Up to now, we only support scanMetrics in iceberg catalog.

**Review by Youngwb — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**